File: SiteVisitsNetCore2020Web/Models/VisitSessionBlock.cs
using SiteVisitsNetCore2020Web.ViewModels;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SiteVisitsNetCore2020Web.Models
{
    public class VisitSessionBlock
    {
        public string Browser { get; set; }
        public string Device { get; set; }
        public List<VisitViewModel> Visits { get; set; }
    }
}

File: SiteVisitsNetCore2020Web/ClientApp/src/app/viewmodels/VisitSessionBlock.ts
//import { VisitInner } from "./VisitInner";

import { Visit } from "../models/Visit";
import { VisitViewModel } from "./VisitViewModel";

export class VisitSessionBlock {
  browser: string;
  device: string;
  visits: Array<VisitViewModel>;

  constructor(options: {
    browser: string, 
    device: string,
    visits: Array<any>
  }) {
    this.browser = options.browser;
    this.device = options.device;
    this.visits = new Array<VisitViewModel>();
    options.visits.forEach(element => this.visits.push(new VisitViewModel(element)));
  }

}
